### DISCUSSION TITLE
What is the advantage/disadvantage of model.set_train_data versus model.load_state_dict?

### DISCUSSION THREAD

**Discussion by PraljakReps** [cornellius-gp/gpytorch#2032] [Q&A] [answered]
I wondered what the advantage/disadvantage between these two approaches for increasing the GP training data: `model.set_train_data(train_x, train_y, strict = False)` and `model.load_state_dict`? The former approach is the direct method of swapping out the training data for the GP. Still, the latter approach seems like a decent alternative by initializing the GP model with the increased training set and using model.load_state_dict to use known GP hyperparameters (for more details of this approach, here is the link: https://botorch.org/v/0.2.2/tutorials/closed_loop_botorch_only).

**Comment by Balandat** [comment, marked as answer]
So using `load_state_dict` you can load the model hyperparameters into a new instance of the model. This allows you to avoid fitting a model (or fitting it from scratch if you want to do some "fine tuning" after adding more data points). 

`set_train_data` instead just updates the training data on an already existing model instance. In that case there is no need to re-load the state dict (as it would only contain the parameters that already on the model).
